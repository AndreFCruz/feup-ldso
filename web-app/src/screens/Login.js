import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Redirect } from 'react-router-dom';
import { Label, Col, Form, FormGroup, Input, Button } from 'reactstrap';
import { login } from '../actions/authenticationActions';
import logo from '../assets/images/impactup.png';

import '../styles/GenericProject.css'; // Importing because of progress button
import '../styles/Login.css';

class Login extends Component {

    constructor(props) {
        super(props);

        this.state = {
            username: '',
            password: ''
        };

        this.handleInputChange = this.handleInputChange.bind(this);
        this.submit = this.submit.bind(this);
    }

    handleInputChange = (event) => {
        this.setState({
            [event.target.name]: event.target.value
        });
    };

    submit() {
        const { login } = this.props;
        const { username, password } = this.state;

        login(username, password);
    }

    render() {
        const { authenticated, loading } = this.props;

        if (!authenticated) {
            return (
                <div className="loginContainer">
                    <div className="cssTool"></div>
                    <div className="login">
                        <div className="logo">
                            <img src={logo} alt="application logo" width="140" />
                        </div>
                        <Form className="form">
                            <Col>
                                <FormGroup>
                                    <Label for="username">Username</Label>
                                    <Input
                                        placeholder="Enter Username"
                                        name="username"
                                        id="username"
                                        value={this.state.username}
                                        onChange={this.handleInputChange} />
                                </FormGroup>
                            </Col>
                            <Col>
                                <FormGroup>
                                    <Label for="username">Password</Label>
                                    <Input
                                        type="password"
                                        placeholder="Password"
                                        name="password"
                                        value={this.state.password}
                                        onChange={this.handleInputChange} />
                                </FormGroup>
                                <Button color="success" className={`loginBtn ${loading ? 'm-progress' : ''}`} onClick={this.submit}>Log in</Button>
                            </Col>
                        </Form>
                    </div>
                </div>
            );
        } else {
            return (
                <Redirect
                    to={{
                        pathname: '/'
                    }}
                />
            );
        }
    }
}

Login.propTypes = {
    authenticated: PropTypes.bool,
    loading: PropTypes.bool,
    login: PropTypes.func
};

const mapStateToProps = ({ auth }) => ({
    authenticated: auth.authenticated,
    loading: auth.loading
});

const mapDispatchToProps = {
    login
};

export default connect(mapStateToProps, mapDispatchToProps)(Login);