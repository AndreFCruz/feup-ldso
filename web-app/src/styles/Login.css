.loginContainer {
  height: 100vh;
  background-image: linear-gradient(var(--secondary-color-light), var(--secondary-color-dark));
  /* background-color: #9e8c5e; */
}

.cssTool {
  height: 8em;
}

.logo {
  text-align: center;
  margin-bottom: 1em;
}

.login {
  opacity: 1;
  width: 350px;
  margin: 0 auto !important;
  margin-top: 600px;
  background-color: var(--primary-color) !important;
  padding: 2em !important;
  border-radius: 0.5em !important;
  box-shadow: 0px 0px 20px 0px rgba(0, 0, 0, 0.3);
}

.loginBtn {
  width: 100%;
}