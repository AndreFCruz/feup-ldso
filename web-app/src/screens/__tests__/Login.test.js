import React from 'react';
import Login from '../Login';
import Enzyme, { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

const middlewares = []; // you can mock any middlewares here if necessary
const mockStore = configureStore(middlewares);

const initialState = {
    auth: {
        authenticated: false,
        loading: false
    },
    login: jest.fn()
};

function setup() {
    const wrapper = shallow(<Login />, {
        context: { store: mockStore(initialState) },
        disableLifecycleMethods: true
    });

    return wrapper;
}


describe('Login screen ', () => {

    it('renders login screen', () => {
        const wrapper = setup();
        expect(wrapper).toMatchSnapshot();
    });

    it('successfully handles submit', () => {

        const wrapper = shallow(
            <Login />, {
                context: { store: mockStore(initialState) },
                disableLifecycleMethods: true
            }
        );


        expect(wrapper).toMatchSnapshot();

        const loginBtn = wrapper.dive().find('Button').first();
        loginBtn.simulate('click');

        expect(wrapper).toMatchSnapshot();
    });
});