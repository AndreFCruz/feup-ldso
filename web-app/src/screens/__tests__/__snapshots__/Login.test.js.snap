// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders without crashing 1`] = `
<div>
  <div
    className="loginContainer"
  >
    <div
      className="cssTool"
    />
    <div
      className="login"
    >
      <div
        className="logo"
      >
        <img
          alt="application logo"
          src="test-file-stub"
          width="140"
        />
      </div>
      <form
        className="form"
      >
        <div
          className="col"
        >
          <div
            className="position-relative form-group"
          >
            <label
              className=""
              htmlFor="username"
            >
              Username
            </label>
            <input
              className="form-control"
              id="username"
              name="username"
              onChange={[Function]}
              placeholder="Enter Username"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="position-relative form-group"
          >
            <label
              className=""
              htmlFor="username"
            >
              Password
            </label>
            <input
              className="form-control"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
          <button
            aria-label={null}
            className="loginBtn  btn btn-success"
            onClick={[Function]}
            type="button"
          >
            Log in
          </button>
        </div>
      </form>
    </div>
  </div>
  <div
    className="notification-container notification-container-empty"
  >
    <span />
  </div>
</div>
`;

exports[`Login screen  renders login screen 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(Login) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "authenticated": false,
      "loading": false,
      "login": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "authenticated": false,
        "loading": false,
        "login": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`Login screen  successfully handles submit 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(Login) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "authenticated": false,
      "loading": false,
      "login": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "authenticated": false,
        "loading": false,
        "login": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`Login screen  successfully handles submit 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(Login) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "authenticated": false,
      "loading": false,
      "login": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "authenticated": false,
        "loading": false,
        "login": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
